AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/InitialSync/SyncEventEmitterTests.swift

```diff
@@ -0,0 +1,188 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import XCTest
+
+@testable import Amplify
+@testable import AmplifyTestCommon
+@testable import AWSPluginsCore
+@testable import AWSDataStoreCategoryPlugin
+
+class SyncEventEmitterTests: XCTestCase {
```

Comment:
So for this test, I am thinking, there are a lot of combination, the one I am testing here is "Post/Comment, isFullSync = true, isDeltaSync = false, added = 1, updated = 0, deleted = 0", should I test for other combination?

Reply:
Ideally, yes. You've got quite a bit of special case logic throughout, and it would be good to validate that logic works as expected.